test/integration/registration/managedcluster_accept_test.go

```diff
@@ -0,0 +1,202 @@
+package registration_test
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+var _ = Describe("ManagedCluster set hubAcceptsClient from true to false", Label("managedcluster-accept"), func() {
+	var managedCluster *clusterv1.ManagedCluster
+	BeforeEach(func() {
+		managedClusterName := fmt.Sprintf("managedcluster-%s", rand.String(6))
+		managedCluster = &clusterv1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: managedClusterName,
+			},
+			Spec: clusterv1.ManagedClusterSpec{
+				HubAcceptsClient: true,
+			},
+		}
+		_, err := clusterClient.ClusterV1().ManagedClusters().Create(context.Background(), managedCluster, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Check rbac files should be created
+		Eventually(func() error {
+			_, err := kubeClient.RbacV1().ClusterRoles().Get(context.Background(), mclClusterRoleName(managedCluster.Name), metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			// check clusterrole is correct
+			clusterRole, err := kubeClient.RbacV1().ClusterRoles().Get(context.Background(), mclClusterRoleName(managedCluster.Name), metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			// Should have rules for certificate rotation
+			hasCertRule := false
+			hasClientCertRule := false
+			hasManagedClusterRule := false
+			hasManagedClusterStatusRule := false
+
+			for _, rule := range clusterRole.Rules {
+				// Check certificate signing request rule
```

Comment:
this looks a bit complicated, and when some rule changes, the test will fail. Can we just read the file from manifests and compare with the output?

Reply:
I have some concerns about this one. I think when a rule changes, the developer should modify the checks here. For example, if the future developer adds the update to the cluster role when accept=false, the integration test case should fail to let the developer know they added a permission that we don't want. 

If we configure a wrong manifest, `reading from manifests and comparing with output` may repeat the mistake twice.

Reply:
If so can we send a SAR instead of comparing the rules?

Reply:
If check every rule is too complicated, how about change to only check on rule when `accept=false`? To make sure when `accept=false`, no `write` permission on `read`

Reply:
I updated to use `DeepEqual`, as we are going to `get` clusterrole anyway,  compare between structure maybe a little simpler than SAR.